app/services/pdf_processor.py

```diff
@@ -0,0 +1,179 @@
+import fitz
+from transformers import pipeline
+from fuzzywuzzy import fuzz
+import json
+
+
+def extract_text_and_coordinates_by_line(pdf_path):
+    """
+    Extracts text and coordinates from the given PDF by line using PyMuPDF.
+
+    Parameters:
+    pdf_path (str): The path to the PDF file.
+
+    Returns:
+    list: A list of dictionaries, each containing the line text and its coordinates.
+    """
+    document = fitz.open(pdf_path)
+    lines_with_coords = []
+
+    for page_num in range(len(document)):
+        page = document[page_num]
+        blocks = page.get_text("dict")["blocks"]
+        for block in blocks:
+            if "lines" in block:
+                for line in block["lines"]:
+                    line_text = ""
+                    line_coords = []
+                    for span in line["spans"]:
+                        line_text += span["text"] + " "
+                        line_coords.append((span["bbox"], span["text"]))
+                    lines_with_coords.append({
```

Comment:
Same comment as before: It will be way easier for you if you create DataClasses to handle complex outputs

```python
class Line(BaseModel)
text: str
coord: Coordinates
page: int

class LineswithCoord(BaseModel):
lines_with_coords: List[Line]
```